lib/variable/variable_instantiate_linalg.cpp

```diff
@@ -30,6 +30,17 @@ constexpr auto structure_element_offset<Eigen::Matrix3d> =
       return offsets.at(key);
     };
 
+// investigate if this can be removed.
```

Comment:
I think we should in fact remove this, if possible.

Reply:
It can't be removed for the moment (fails to compile) - but I thought you were looking at implementing a different mechanism for access to structured data from these types, separately to this work?

Reply:
As described in the issue thread what we had discussed is unfortunately not possible: https://github.com/scipp/scipp/issues/2252#issuecomment-960502687. Can we throw here instead?